srcjs/input_binding_fileinput.js

```diff
@@ -177,6 +177,8 @@ $.extend(FileUploader.prototype, FileProcessor.prototype);
   };
   this.$setError = function(error) {
     this.$bar().toggleClass('progress-bar-danger', (error !== null));
+    // bs4 forward compatibility
+    this.$bar().toggleClass('bg-danger', (error !== null));
```

Comment:
It's weird that there isn't a `text-danger`, isn't it? https://getbootstrap.com/docs/4.0/utilities/colors/#background-color

Maybe we should set `progress-bar-danger` in bs3compat, using `color-yiq($danger)` or whatever to set the `color`; or else maybe we should introduce `.text-danger`.

Reply:
https://github.com/rstudio/bootstraplib/pull/30